src/libs/actions/IOU.js

```diff
@@ -722,11 +723,228 @@ function payIOUReport({
     return promiseWithHandlers;
 }
 
+/**
+ * @param {String} chatReportID - Report ID of the chat where the IOU is.
+ * @param {Number} amount - IOU amount in cents.
+ * @param {String} currency - IOU currency.
+ * @param {String} comment
+ * @param {Array} participants - Participants on the IOU
+ */
+function sendMoneyWithWallet(chatReportID, amount, currency, comment, participants) {
+    const chatReport = _.empty(chatReportID) ? ReportUtils.buildOptimisticChatReport(participants) : chatReports[`${ONYXKEYS.COLLECTION.REPORT}${chatReportID}`];
+    const ownerEmail = participants[0].login;
+    const userEmail = participants[1].login;
+    const optimisticIOUReport = ReportUtils.buildOptimisticIOUReport(
+        ownerEmail,
+        userEmail,
+        amount,
+        chatReport.chatReportID,
+        currency,
+        preferredLocale,
+    );
+    const newSequenceNumber = Report.getMaxSequenceNumber(chatReport.reportID) + 1;
+    const optimisticIOUReportAction = ReportUtils.buildOptimisticIOUReportAction(
+        newSequenceNumber,
+        CONST.IOU.REPORT_ACTION_TYPE.PAY,
+        amount,
+        currency,
+        comment,
+        participants,
+        CONST.IOU.PAYMENT_TYPE.EXPENSIFY,
+        '',
+        optimisticIOUReport.reportID,
+    );
+    const optimisticData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReport.chatReportID}`,
+            value: {
+                [optimisticIOUReportAction.reportActionID]: {
+                    optimisticIOUReportAction,
+
+                    // Set pendingAction state for pattern B
+                    pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+                },
+            },
+        },
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${chatReport.chatReportID}`,
```

Comment:
I think this is reportID https://github.com/Expensify/App/blob/9459789c8c2dcfd75adc0619b84157a24be19f56/src/libs/ReportUtils.js#L837-L859
```suggestion
            key: `${ONYXKEYS.COLLECTION.REPORT}${chatReport.reportID}`,
```